packages/node/src/builders/package/utils/update-package-json.ts

```diff
@@ -8,7 +8,7 @@ export default function updatePackageJson(
   options: NormalizedBuilderOptions,
   context: BuilderContext
 ) {
-  const mainFile = basename(options.main, '.ts');
+  const mainFile = basename(options.main).replace(/\.[tj]s$/, '');
```

Comment:
Is it okay if there's no extension in the `package.json`?

Reply:
The extensions are added on the lines below this

Reply:
Or do you mean, does this work if there is no `.ts` or `.js` extension?
I only removes the extension if the string ends with `.ts` or `.js`